interfaces/udev/backend.go

```diff
@@ -153,15 +153,19 @@ func (b *Backend) Setup(appSet *interfaces.SnapAppSet, opts interfaces.Confineme
 
 	var deviceBuf bytes.Buffer
 	deviceBuf.WriteString("# This file is automatically generated.\n")
-	deviceBuf.WriteString("# snap is allowed to manage own device cgroup.\n")
 
 	if udevSpec.ControlsDeviceCgroup() {
 		// The spec states that the snap can manage its own device
 		// cgroup (typically applies to container-like snaps), in which
 		// case leave a flag for snap-confine in at a known location.
-
+		deviceBuf.WriteString("# snap is allowed to manage own device cgroup.\n")
 		deviceBuf.WriteString("self-managed=true\n")
 	}
+	if (opts.DevMode || opts.Classic) && !opts.JailMode {
```

Comment:
Perhaps classic should be distinct from devmode.

Reply:
This duplicates the behavior we have right here: https://github.com/snapcore/snapd/pull/13777/files/946315b92797f4bc66b0d9aed23b9f8b89481f09#diff-0aa4a229d9fd98c2fba19a47254f6e2ebf8159111468122090d63741a6598fbaR115

Reply:
but looking there, one could argue then that the flag shouldn't be devmode=true but nonstrict=true then?

Reply:
I was thinking about the name of the flag, it's not devmode when snap is classic. I agree it should be something else than now.

Reply:
changed to `non-strict=true`